### PR TITLE
googleAuthenticator: 1.0 -> 1.03

### PR DESCRIPTION
**Pull request by aneeshusa**
###### Motivation for this change

A new release (version 1.03) of google-authenticator-libpam has been cut; this is a follow up from #19603.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/google-authenticator/default.nix [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
+    url = "https://github.com/google/google-authenticator-libpam/archive/${version}.tar.gz";
+    sha256 = "0wb95z5v1w4sk0p7y9pbn4v95w9hrbf80vw9k2z2sgs0156ljkb7";
   };
 
-  buildInputs = [ pam ];
+  buildInputs = [ autoreconfHook pam ];
 
   preConfigure = ''
-    sed -i 's|libqrencode.so.3|${qrencode}/lib/libqrencode.so.3|' google-authenticator.c
+    sed -i "s|libqrencode.so.3|$qrencode/lib/libqrencode.so.3|" src/google-authenticator.c
```
Comment (grahamc) [inline pkgs/os-specific/linux/google-authenticator/default.nix]: why this change? I'm not sure $qrencode will work there, like that?

Comment (aneeshusa) [inline pkgs/os-specific/linux/google-authenticator/default.nix]: I like to use environment variables where possible over Nix antiquotations. Here it doesn't make a difference. However, with `version` and other values that can't be `override`n easily, it makes it easier to call `updateAttrs` since the old value of e.g. `version` isn't baked into the various shell snippets.

Comment (grahamc) [inline pkgs/os-specific/linux/google-authenticator/default.nix]: Yes, but `qrencode` isn't an environment variable here.

`set -u` before this errors:

```
/nix/store/gqkczsmhc596isabbl5p2dw1i5rmwqks-stdenv/setup: line 70: qrencode: unbound variable
```

**Review by grahamc — CHANGES_REQUESTED** [review]

**Comment by aneeshusa** [comment]
@grahamc I've fixed that, guess I wasn't thinking so straight at 4:30 AM :)

**Comment by grahamc** [comment]
Merged in 652a87018b9c86e02d7c7b5c627a19bd7b73543c thank you!
